commands/transaction.go

```diff
@@ -30,6 +34,34 @@ func NewInputData(height uint64) *commandspb.InputData {
 	}
 }
 
+func MarshalInputData(chainID string, inputData *commandspb.InputData) ([]byte, error) {
+	data, err := proto.Marshal(inputData)
+	if err != nil {
+		return nil, err
+	}
+
+	dataWithChainID := append([]byte(fmt.Sprintf("%s%c", chainID, ChainIDDelimiter)), data...)
+
+	return dataWithChainID, nil
+}
+
+func UnmarshalInputData(rawInputData []byte, expectedChainID string) (*commandspb.InputData, error) {
+	prefix := fmt.Sprintf("%s%c", expectedChainID, ChainIDDelimiter)
```

Comment:
Do we not want to use the transaction version somewhere to tell us whether we should even be looking to look for a pre-fixed chainID or not?

Reply:
yes it should be used in the core as well I suppose? For now we should keep supporting both the version 2 and 3 just until everyone upgrade their wallet?